File: Python/problem65.py
def sumofdigits(a):
	if a < 10:
		return a;
	return a % 10 + sumofdigits(a/10);

def main():
	x = 100;
	n = 1;
	d = 0;
	if (x-2) % 3 == 0 or (x-2) % 3 == 2:
		d = 1;
	else:
		d = 2 * ((x-2)/3+1);
	for e in range(0,x-2):
		g = (x-3)-e
		if g % 3 == 0 or g % 3 == 2:
			f = 1;
		else:
			f = 2 * (g/3+1);
		n += f*d;
		temp = d;
		d = n;
		n = temp;
	n += 2*d;
	print sumofdigits(n);

if __name__ == "__main__":
	main()

File: problem65.cpp
#include <iostream>
#include <math.h>

using namespace std;

int sumofdigits(int a) {
	if (a < 10) {
		return a;
	}
	return a % 10 + sumofdigits(a/10);
}

void swap(int& a, int& b) {
	int temp = a;
	a = b;
	b = temp;
}

int main() {
	for (int j = 3; j <= 10; j++) {
		const int x = j;
		int n = 1;
		int d = 0;
		if ((x-2) % 3 == 0 || (x-2) % 3 == 2)
			d = 1;
		else
			d = 2 * ((x-2)/3+1);
		int esum = 0;
		for (int e = x-3; e >= 0; e--) {
			esum += e;
			int f;
			if (e % 3 == 0 || e % 3 == 2)
				f = 1;
			else
				f = 2 * (e/3+1);
			n += f*d;
			swap(d, n);
		}
		cout << d<< endl;
		//cout << j;
		n += 2*d;
		//cout << n << '/' << d << endl;
	}
	return 0;
}